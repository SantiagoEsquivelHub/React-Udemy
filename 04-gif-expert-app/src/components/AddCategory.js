import React, { useState } from 'react'
import PropTypes from 'prop-types'


export const AddCategory = ({ setCategories }) => {

    const [inputValue, setInputValue] = useState('');

    const handleCategory = (e) => {
        setInputValue(e.target.value);
        console.log('handleCategory llamado')
    }

    const handleSubmit = (e) => {
        e.preventDefault(); //Evita el refresh de la pagina al hacer sl submit
        console.log('handleSubmit llamado')

        if (inputValue.trim().length > 2) {
            setCategories(cats => [inputValue, ...cats]);
            setInputValue('');
        }
    }

    return (


        <form onSubmit={handleSubmit}>
            <p>{inputValue}</p>
            <input
                type="text"
                value={inputValue}
                onChange={handleCategory}
            />
        </form>



    )
}

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}

