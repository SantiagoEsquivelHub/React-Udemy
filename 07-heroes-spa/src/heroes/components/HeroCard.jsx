import { Link } from "react-router-dom";

export const HeroCard = ({
    id,
    superhero,
    alter_ego,
    first_appearance,
    characters,
}) => {

    const CharactersByHero = ({ alter_ego, characters }) => {

        if (alter_ego !== characters) {
            return (<p>{characters}</p>)
        }
    }

    const heroeImgUrl = `/assets/heroes/${id}.jpg`;
    //const charactersByHero = (<p>{characters}</p>);

    return (
        <div className="col">
            <div className="card">
                <div className="row no-gutters">
                    <div className="col-4">
                        <img src={heroeImgUrl} className="card-img" alt={superhero} />
                    </div>

                    <div className="col-8">
                        <div className="card-body">
                            <h5 className="card-title">{superhero}</h5>
                            <p className="card-text">{alter_ego}</p>
                            <CharactersByHero alter_ego={alter_ego} characters={characters} />
                            <p className="card-text text-muted">{first_appearance}</p>
                            <Link to={`/hero/${id}`}>
                                MÃ¡s...
                            </Link>
                        </div>
                    </div>
                </div>
            </div>
        </div>
    )
}
