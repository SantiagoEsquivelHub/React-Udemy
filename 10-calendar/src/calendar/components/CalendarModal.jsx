import { useEffect, useMemo, useState } from 'react';
import { addHours } from 'date-fns';
import { useCalendarStore, useForm, useUiStore } from '../../hooks';
import Swal from 'sweetalert2';
import 'sweetalert2/dist/sweetalert2.min.css';
import DatePicker, { registerLocale } from 'react-datepicker';
import Modal from 'react-modal';
import '../../styles.css';
import 'react-datepicker/dist/react-datepicker.css';
import esES from 'date-fns/locale/es'
import { differenceInSeconds } from 'date-fns/esm';

registerLocale('es', esES)

const customStyles = {
    content: {
        top: '50%',
        left: '50%',
        right: 'auto',
        bottom: 'auto',
        marginRight: '-50%',
        transform: 'translate(-50%, -50%)',
    },
};

// Make sure to bind modal to your appElement (https://reactcommunity.org/react-modal/accessibility/)
Modal.setAppElement('#root');

export const CalendarModal = () => {

    const [formSubmitted, setFormSubmitted] = useState(false);

    const { title, notes, start, end, onInputChange, onDateChanged, formState, onEventSelected } = useForm({
        title: '',
        notes: '',
        start: new Date(),
        end: addHours(new Date(), 2)
    });

    const { isDateModalOpen, closeDateModal } = useUiStore();
    const { activeEvent, startSavingEvent } = useCalendarStore();

    const titleClass = useMemo(() => {

        if (!formSubmitted) return '';

        return title.length > 0
            ? ''
            : 'is-invalid'

    }, [title, formSubmitted])

    useEffect(() => {
        if (!!activeEvent) {
            onEventSelected({ ...activeEvent });
        }
    }, [activeEvent])


    const onSubmit = async (event) => {

        event.preventDefault();
        setFormSubmitted(true);

        const difference = differenceInSeconds(end, start);

        if (isNaN(difference) || difference <= 0) {
            Swal.fire('Fechas incorrectas', 'Revisar las fechas ingresadass');
            return;
        }

        if (title <= 0) {
            Swal.fire('Fechas incorrectas', 'Revisar las fechas ingresadass');
            return;
        }

        console.log(formState)

        await startSavingEvent(formState);
        closeDateModal();
        setFormSubmitted(false);
    }

    return (
        <Modal
            className='modal'
            overlayClassName='modal-fondo'
            isOpen={isDateModalOpen}
            onRequestClose={closeDateModal}
            style={customStyles}
            contentLabel="Example Modal"
        >
            <h1> Nuevo evento </h1>
            <hr />
            <form
                onSubmit={onSubmit}
                className="container">

                <div className="form-group mb-2">
                    <label>Fecha y hora inicio</label>
                    <DatePicker
                        className='form-control'
                        onChange={(e) => onDateChanged(e, 'start')}
                        selected={start}
                        dateFormat='Pp'
                        showTimeSelect
                        locale='es'
                        timeCaption='Hora'
                    />
                </div>

                <div className="form-group mb-2">
                    <label>Fecha y hora fin</label>
                    <DatePicker
                        minDate={start}
                        className='form-control'
                        onChange={(e) => onDateChanged(e, 'end')}
                        selected={end}
                        dateFormat='Pp'
                        showTimeSelect
                        locale='es'
                        timeCaption='Hora'
                    />
                </div>

                <hr />
                <div className="form-group mb-2">
                    <label>Titulo y notas</label>
                    <input
                        type="text"
                        className={`form-control ${titleClass}`}
                        placeholder="Título del evento"
                        name="title"
                        value={title}
                        onChange={onInputChange}
                        autoComplete="off"
                    />
                    <small id="emailHelp" className="form-text text-muted">Una descripción corta</small>
                </div>

                <div className="form-group mb-2">
                    <textarea
                        type="text"
                        className="form-control"
                        placeholder="Notas"
                        rows="5"
                        name="notes"
                        value={notes}
                        onChange={onInputChange}
                    ></textarea>
                    <small id="emailHelp" className="form-text text-muted">Información adicional</small>
                </div>

                <button
                    type="submit"
                    className="btn btn-outline-primary btn-block"
                >
                    <i className="far fa-save"></i>
                    <span> Guardar</span>
                </button>

            </form>

        </Modal>
    )
}
