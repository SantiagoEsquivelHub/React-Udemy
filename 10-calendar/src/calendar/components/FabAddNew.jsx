import { addHours } from 'date-fns';
import { useCalendarStore, useUiStore } from '../../hooks';
import '../../styles.css';

export const FabAddNew = () => {

    const { openDateModal } = useUiStore();
    const { setActiveEvent } = useCalendarStore();


    const handleClickNew = () => {
        setActiveEvent({
            title: '',
            notes: '',
            start: new Date(),
            end: addHours(new Date(), 2),
            user: {
                _id: 'ABC123',
                name: 'Santiago'
            }
        })
        openDateModal();
    }

    return (
        <button
            className="btn btn-primary fab"
            onClick={handleClickNew}
        >
            <i className='fas fa-plus'></i>
        </button>
    )
}
