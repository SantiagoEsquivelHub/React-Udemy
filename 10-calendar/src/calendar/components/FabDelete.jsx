import { useCalendarStore } from '../../hooks';
import '../../styles.css';

export const FabDelete = () => {

    const { hasEventSelected, startDeletingEvent } = useCalendarStore();


    const handleDelete = async () => {
        await startDeletingEvent();
    }

    return (
        <button
            className="btn btn-danger fab-danger"
            onClick={handleDelete}
            aria-label="btn-delete"
            style={{
                display: hasEventSelected ? '' : 'none'
            }}
        >
            <i className='fas fa-trash-alt'></i>
        </button>
    )
}
