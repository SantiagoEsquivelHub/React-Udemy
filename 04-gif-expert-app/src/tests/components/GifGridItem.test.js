
import '@testing-library/jest-dom';  // importacion del jest para snippets
import { shallow } from 'enzyme'; // importacion del enzyme
import { GifGridItem } from '../../components/GifGridItem';

describe('Pruebas del componente GifGridItem', () => {

    const title = 'Un titulo';
    const url = 'https://localhost/algo.jpg';
    const wrapper = shallow(<GifGridItem title={title} url={url} />);


    test('Debe mostrar el componente correctamente', () => {

        expect(wrapper).toMatchSnapshot();
    });

    test('Debe tener un parrafo con el title', () => {

        const parrafo = wrapper.find('p');
        expect(parrafo.text().trim()).toBe(title);

    });

    test('Debe tener la imagen igual al url y alt de los props', () => {

        const img = wrapper.find('img');
        /* console.log(img.prop('src')); */
        expect(img.prop('src')).toBe(url);
        expect(img.prop('alt')).toBe(title);

    });

    test('Debe tener la imagen igual al url y alt de los props', () => {

        const div = wrapper.find('div');
        const className = div.prop('className');
         console.log(div.prop('className'));
        expect(className.includes("animate__fadeIn")).toBe(true);

    });
})