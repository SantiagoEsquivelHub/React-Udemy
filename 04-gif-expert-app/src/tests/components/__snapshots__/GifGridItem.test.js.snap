// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas del componente GifGridItem Debe mostrar el componente correctamente 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <p>
    Un titulo
  </p>
  <img
    alt="Un titulo"
    src="https://localhost/algo.jpg"
  />
</div>
`;
